docs/langref/relay_expr.rst

```diff
@@ -0,0 +1,527 @@
+- `If-Then-Else`_ Expressions
+- Recursive Calls in Functions
+
+Because a function can be viewed as a subgraph from the point of view of computational graphs,
+a call effectively substitutes in the subgraph. Thus, if a function's body is entirely within
+the dataflow fragment, a call to that function is in the dataflow fragment; conversely, if the
+function's body contains control flow, a call to that function is not part of the dataflow fragment.
+
+Variables
+=========
+
+Inspired by LLVM, Relay explicitly distinguishes between local and
+global variables both in the AST and in the text format. In the text format,
+global and local variables are distinguished by prefixes, or sigils.
+Global variables are prefixed with :code:`@` and local variables with :code:`%`.
+
+This explicit distinction makes certain optimizations easier to implement.
+For example, inlining a global definition requires no analysis: simply
+substituting the definition suffices.
+
+Global Variable
+~~~~~~~~~~~~~~~~~~
+
+Global identifiers are prefixed by the :code:`@` sigil, such as ":code:`@global`".
+A global identifier always references a globally visible definition contained in the environment.
+Global identifiers must be unique.
+
+See :py:class:`~tvm.relay.expr.GlobalVar` for its implementation
+and documentation.
+
+Local Variable
+~~~~~~~~~~~~~~
+
+Local identifiers are prefixed by the :code:`%` sigil,
+such as ":code:`%local`". A local identifier always references
+a function argument or a variable bound in a :code:`let` expression,
+and will be scoped to the function where it appears or the :code:`let`
+expression where it is bound, respectively.
+
+In the below code segment, notice that :code:`%a` is defined twice. This is
+permitted, as in most functional languages; in the scope of the second
+:code:`let` expression, the name :code:`%a` is "shadowed," meaning all
+references to :code:`%a` in the inner scope refer to the later defintion, while
+references to :code:`%a` in the outer scope continue to refer to
+the first one.
+
+.. code-block:: python
+let %a = 1;
+let %b = 2 * %a;  // %b = 2
+let %a = %a + %a; // %a = 2. %a is shadowed
+%a + %b           // has value 2 + 2 = 4
+
+(Note that in Relay's implementation, each definition of a local variable
+creates a new :py:class:`~tvm.relay.expr.Var`, so a shadowed local variable,
+despite having the same name as one in an outer scope, will be a different
+object. This allows for comparing local variables by pointer identity with the
+knowledge that the same local variable object corresponds to a different binding site.)
+
+See :py:class:`~tvm.relay.expr.Var` for its implementation
+and documentation.
+
+Functions
+=========
+
+Functions in Relay act similarly to procedures or functions in
+other programming languages and serve to generalize the concept
+of a named subgraph.
+
+See :py:class:`~tvm.relay.expr.Function` for the definition and documentation of function nodes.
+
+Global Functions
+~~~~~~~~~~~~~~~~
+
```

Comment:
Let us add a section about Module, or rename Global Function into Module. 

as per @reminisce 's suggestion, mention that the Module mechanism is naturally used to represent subgraphs in traditional deep learning framework's terminology. Give an example of a function calling into sub functions

Reply:
Right, will do.